File: concrete/src/test/java/pcb/uwu/amount/quantity/DurationTest.java
package pcb.uwu.amount.quantity;

import org.junit.Test;

import java.time.Duration;

import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.quantity.HourUnit.HOUR;
import static pcb.uwu.unit.quantity.SecondUnit.SECOND;

public class DurationTest {
	@Test
	public void testConstructFromDuration() {
		assertEquals("0.000000001 s", new Time(Duration.ofNanos(1), SECOND).toString());
		assertEquals("0.001 s", new Time(Duration.ofMillis(1), SECOND).toString());
		assertEquals("1 s", new Time(Duration.ofSeconds(1), SECOND).toString());
		assertEquals("60 s", new Time(Duration.ofMinutes(1), SECOND).toString());
		assertEquals("3600 s", new Time(Duration.ofHours(1), SECOND).toString());
		assertEquals("86400 s", new Time(Duration.ofDays(1), SECOND).toString());
		assertEquals("24 h", new Time(Duration.ofDays(1), HOUR).toString());
	}
}

File: concrete/src/test/java/pcb/uwu/amount/base/DurationTest.kt
package pcb.uwu.amount.base

import org.junit.Assert.assertEquals
import org.junit.Test
import pcb.uwu.unit.base.HourUnit
import pcb.uwu.unit.base.SecondUnit
import java.time.Duration

class DurationTest
{
    @Test
    fun testConstructFromDuration()
    {
        assertEquals("0.000000001 s", Time(Duration.ofNanos(1), SecondUnit).toString())
        assertEquals("0.001 s", Time(Duration.ofMillis(1), SecondUnit).toString())
        assertEquals("1 s", Time(Duration.ofSeconds(1), SecondUnit).toString())
        assertEquals("60 s", Time(Duration.ofMinutes(1), SecondUnit).toString())
        assertEquals("3600 s", Time(Duration.ofHours(1), SecondUnit).toString())
        assertEquals("86400 s", Time(Duration.ofDays(1), SecondUnit).toString())
        assertEquals("24 h", Time(Duration.ofDays(1), HourUnit).toString())
    }
}